src/main/java/org/folio/rest/impl/InventoryHelper.java

```diff
@@ -659,18 +661,25 @@ private JsonObject getFirstObjectFromResponse(JsonObject response, String proper
    * @return value from cache
    */
   private CompletableFuture<JsonObject> getAndCache(String key, UnaryOperator<String> endpointConstructor, Function<JsonObject, JsonObject> fn) {
-    JsonObject response = ctx.get(key);
+    String tenantSpecificKey = buildTenantSpecificKey(key);
+    JsonObject response = ctx.get(tenantSpecificKey);
+
     if(response == null) {
       return handleGetRequest(endpointConstructor.apply(key), httpClient, ctx, okapiHeaders, logger).thenApply(json -> {
         JsonObject result = fn.apply(json);
-        ctx.put(key, result);
+        ctx.put(tenantSpecificKey, result);
```

Comment:
I don' think using Context for caching is a viable option. If I understand it correctly the context is created when the verticle is first deployed. Say the instance-type and instance-status are created after an initial call to orders, the context object will forever have the empty values in the cache which it retrieved the first time and will not retrieve the newly created entries.

Reply:
If, for example, there is no required instance-type in the context, this method will try to get it from the inventory-storage and, if successful, will put the result in context.

Reply:
I got what you meant. I've reworked it so that objects with empty values aren't cached.